### DISCUSSION TITLE
textedit with ScrollArea::both

### DISCUSSION THREAD

**Discussion by pm100** [emilk/egui#4871] [Q&A]
Trying to make a multiline textedit that does not wrap lines but scrolls horizontally.

I have

 ```
       egui::ScrollArea::both().show(ui, |ui| {
            ui.add_sized(ui.available_size(), |ui: &mut Ui| {
                let widget = egui::TextEdit::multiline(&mut self.text)
                    .font(egui::TextStyle::Monospace)
                    .code_editor()
                    .lock_focus(true)
                    .desired_width(0.0f32)
                    // .desired_width(f32::INFINITY)
                    .id(self.id);
```

(You can see me guessing at the tweaks I need)

If a type a line that goes off the visible area it wraps rather than forcing a horizontal scroll bar. (Vertical works fine)

**Comment by sugarflowers** [comment]
How about this?

```rs
let theme = egui_extras::syntax_highlighting::CodeTheme::from_memory(ui.ctx());
let mut layouter = |ui: &egui::Ui, string: &str, wrap_width: f32| {
    let layout_job =  egui_extras::syntax_highlighting::highlight(ui.ctx(), &theme, string, "".into());
    ui.fonts(|f| f.layout_job(layout_job))
};

egui::ScrollArea::both()
.auto_shrink(false)
.show(ui, |ui| {
   ui.add_enabled(true, egui::TextEdit::multiline(&mut self.text)
        .code_editor()
        .layouter(&mut layouter)
        .desired_rows(10)
        .desired_width(f32::INFINITY)
        .frame(false)
    );
});
```

By the way, "".into() is a code syntax specification and is defined like this in the egui source code.
If you change it to something like "rs".into() then rust syntax highlighting will be enabled.

```rs
        match language.to_lowercase().as_str() {
            "c" | "h" | "hpp" | "cpp" | "c++" => Some(Self::cpp()),
            "py" | "python" => Some(Self::python()),
            "rs" | "rust" => Some(Self::rust()),
            "toml" => Some(Self::toml()),
            _ => {
                None // unsupported language
            }
```

**Reply by pm100** [reply]
i do not want syntax highlighting, which bit of this actually makes it scroll horizontally

**Reply by sugarflowers** [reply]
In the sample, an empty string is specified so that syntax highlighting will not work.
And after trying various things, I believe that horizontal scrolling is achieved by the layouter.
Therefore, the code makes use of the layouter even though syntax highlighting is not required.
Of course, there may be a better way.
